pandas/io/formats/style.py

```diff
@@ -204,7 +198,11 @@ def __init__(
         self.tooltips: Optional[_Tooltips] = None
         self._display_funcs: DefaultDict[  # maps (row, col) -> formatting function
             Tuple[int, int], Callable[[Any], str]
-        ] = defaultdict(lambda: self._default_display_func)
+        ] = defaultdict(lambda: partial(_default_formatter, precision=None))
+        self.precision = precision  # can be removed on set_precision depr cycle
+        self.na_rep = na_rep  # can be removed on set_na_rep depr cycle
+        if any((precision is not None, na_rep is not None)):
```

Comment:
i would just call `self.format(....)` here and have formatter check if any args are not None before doing anything (otherwise a no-op)

Reply:
implemented a clear procedure - serves dual purposes and is efficient.